workflows/rnaseq-ref-index/make_pre_mrna_fasta.R

```diff
@@ -93,45 +99,69 @@ if (!dir.exists(opt$annotation_output)) {
 }
 
 
-# extract GRanges object containing genomic coordinates of each 
-# annotated transcript - both spliced transcripts and intronic regions
+# extract GRanges object containing genomic coordinates of each annotated transcript
+# both spliced transcripts and intronic regions
 grl <- eisaR::getFeatureRanges(
   gtf = opt$gtf,
   featureType = c("spliced", "intron"), 
+  flankLength = 86L,
+  joinOverlappingIntrons = FALSE,
   verbose = TRUE
 )
 
 # load in primary genome
 genome <- Biostrings::readDNAStringSet(opt$genome)
 names(genome) <- stringr::word(names(genome), 1)
 
-# extract spliced and intron sequences
-# genomic regions defined above
-seqs <- GenomicFeatures::extractTranscriptSeqs(
+
+# add the levels and lengths
+seqlevels(grl) <- seqlevels(genome)
+seqlengths(grl) <- seqlengths(genome)
+
+# trim grl used for splici
+splici_grl <- trim(grl)
```

Comment:
I don't think there is any major harm the way this is, but if we never use the bare `grl` after this point (we are not keeping two version around anymore) it might make sense to revert this to just `grl` to save memory.

Reply:
So we actually do still use the regular grl. When you trim it, you lose the metadata from the grl, very unclear why. And until the USA mode was released, I had been grabbing the metadata (which maps the spliced gene to the intron gene id) and using that to collapse the counts. I don't want to get rid of that step just yet since I haven't actually had the change to play around with using the USA output.